src/Http/Http/src/Internal/DefaultHttpRequest.cs

```diff
@@ -147,10 +147,7 @@ public override IRequestCookieCollection Cookies
         set { Headers.ContentType = value; }
     }
 
-    public override bool HasFormContentType
-    {
-        get { return FormFeature.HasFormContentType; }
-    }
+    public override bool HasFormContentType => HttpExtensions.HasApplicationFormContentType(ContentType) || HttpExtensions.HasMultipartFormContentType(ContentType);
```

Comment:
Use new `HasFormContentType(...)` method here.

Reply:
This is a behavior change mentioned here: https://github.com/dotnet/aspnetcore/pull/50016#discussion_r1291538306
Before if the `FormFeature.Form` was set manually this would also return `true` independent of the headers.
Now it would return `false` because only the headers are checked.
`DefaultHttpRequest.HasFormContentType` needs to be exactly the same as `FormFeature.HasFormContentType` for backward compatibility.

I guess this should also check if the form is already set e.g.

```c#
_features.Cache.Form?._form is not null || ...
```

Reply:
> Now it would return false because only the headers are checked.
DefaultHttpRequest.HasFormContentType needs to be exactly the same as FormFeature.HasFormContentType for backward compatibility.

Yeah, this is a total drag. I almost considered a breaking change here. However, it seems like there's a few components in MVC's model binding that rely on the behavior of form set = form content type is true. At best, here we can avoid constructing a new `FormFeature` if it has already been set.